src/hotspot/share/opto/compile.cpp

```diff
@@ -3620,15 +3620,13 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
 #endif
 
   case Op_ModI:
-    if (UseDivMod) {
+    if (UseDivMod && !n->has_prec_edges()) {
       // Check if a%b and a/b both exist
       Node* d = n->find_similar(Op_DivI);
-      if (d) {
+      if (d && !d->has_prec_edges()) {
```

Comment:
Could be replaced with
```suggestion
      if (d != nullptr && !d->has_prec_edges()) {
```
Same at other places below.